ci_scripts/meson-build.sh

```diff
@@ -17,5 +17,31 @@ source "$SCRIPT_DIR/env.sh"
 
 cd "$SCRIPT_DIR/.."
 
-meson setup build --prefix "$INSTALL_DIR" --buildtype="$1" -Dcassert=true -Dtap_tests=enabled $ENABLE_COVERAGE
+BUILD_TYPE=
+
+case "$1" in
+    debug)
+        echo "Building with debug option"
+        BUILD_TYPE=$1
+        ;;
+
+    debugoptimized)
+        echo "Building with debugoptimized option"
+        BUILD_TYPE=$1
+        ;;
+
+    sanitize)
+        echo "Building with sanitize option"
+        export CFLAGS="-fsanitize=address -fsanitize=undefined -fno-omit-frame-pointer -fno-inline-functions"
+        BUILD_TYPE=debug
```

Comment:
What is the best practices about optimization and sanitizers? Our production build will be `-O2` but maybe that fucks up sanitizers.

Reply:
I didn't see information that `-O2` could harm sanitizer checks, however lower optimization levels simplify report reviews. I hope there should be no difference between `-O0` and `-O2` in terms of leaks and UBs, am I wrong?